datatypes/src/primitives/bounding_box.rs

```diff
@@ -357,7 +357,7 @@ impl BoundingBox2D {
     /// ```
     ///
     pub fn intersection(&self, other_bbox: &Self) -> Option<Self> {
-        if self.overlaps_bbox(other_bbox) {
+        if self.intersects_bbox(other_bbox) {
```

Comment:
Could you add a test case that failed before and is now correct?

Reply:
added intersects_inner() test